### PR TITLE
[Admin] Setup a space directly from front-office

### PR DESCRIPTION
**Pull request by NicolasEYSSERIC**
closing feature #8899

### CODE REVIEW

File: core-war/src/main/java/org/silverpeas/web/jobstartpage/control/JobStartPagePeasSessionController.java [hunk 1]
```diff
@@ -199,11 +199,21 @@ public void setSubSpaceId(String subSpaceId) {
   }
 
   public void setSpaceId(String spaceUserId) {
-    if (m_NavBarMgr.setCurrentSpace(spaceUserId)) {
-      setManagedSpaceId(spaceUserId, true);
+    String spaceId = spaceUserId;
+    if (StringUtil.isDefined(spaceId)) {
+      List<SpaceInstLight> path = ServiceProvider.getService(AdminController.class).getPathToSpace(spaceId, true);
+      if (path.size() >= 2) {
```
Comment (silveruser) [inline core-war/src/main/java/org/silverpeas/web/jobstartpage/control/JobStartPagePeasSessionController.java]: MINOR Assign this magic number 2 to a well-named constant, and use the constant instead. [rule](http://integration.silverpeas.org/sonar/coding_rules#rule_key=squid%3AS109)

**Comment by silveruser** [comment]
SonarQube analysis reported 2 issues
* ![MAJOR][MAJOR] 1 major
* ![MINOR][MINOR] 1 minor

Watch the comments in this conversation to review them.

#### 1 extra issue

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [JobManagerPeasRequestRouter.java#L76](https://github.com/Silverpeas/Silverpeas-Core/blob/ee7e2896c6efedadf4c510338e184faccca425aa/core-war/src/main/java/org/silverpeas/web/jobmanager/servlets/JobManagerPeasRequestRouter.java#L76): The Cyclomatic Complexity of this method "getDestination" is 11 which is greater than 10 authorized. [rule](http://integration.silverpeas.org/sonar/coding_rules#rule_key=squid%3AMethodCyclomaticComplexity)

[MAJOR]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR'
[MINOR]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR'

**Comment by silveruser** [comment]
SonarQube analysis reported 1 issue

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [JobManagerPeasRequestRouter.java#L76](https://github.com/Silverpeas/Silverpeas-Core/blob/5c3541d8783c19f94efea728131edeb34cd26f7b/core-war/src/main/java/org/silverpeas/web/jobmanager/servlets/JobManagerPeasRequestRouter.java#L76): The Cyclomatic Complexity of this method "getDestination" is 11 which is greater than 10 authorized. [rule](http://integration.silverpeas.org/sonar/coding_rules#rule_key=squid%3AMethodCyclomaticComplexity)

[MAJOR]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR'

File: core-web/src/main/java/org/silverpeas/core/web/util/viewgenerator/html/window/AbstractWindow.java [hunk 2]
```diff
@@ -32,10 +32,12 @@
 import org.silverpeas.core.cache.service.CacheServiceProvider;
 import org.silverpeas.core.template.SilverpeasTemplate;
 import org.silverpeas.core.template.SilverpeasTemplateFactory;
+import org.silverpeas.core.util.ArrayUtil;
 import org.silverpeas.core.util.LocalizationBundle;
 import org.silverpeas.core.util.ResourceLocator;
 import org.silverpeas.core.util.StringUtil;
 import org.silverpeas.core.util.URLUtil;
```
Comment (silveruser) [inline core-web/src/main/java/org/silverpeas/core/web/util/viewgenerator/html/window/AbstractWindow.java]: MINOR Remove this unused import 'org.silverpeas.core.util.URLUtil'. [rule](http://integration.silverpeas.org/sonar/coding_rules#rule_key=squid%3AUselessImportCheck)

**Comment by silveruser** [comment]
SonarQube analysis reported 2 issues
* ![MAJOR][MAJOR] 1 major
* ![MINOR][MINOR] 1 minor

Watch the comments in this conversation to review them.

#### 1 extra issue

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [JobManagerPeasRequestRouter.java#L76](https://github.com/Silverpeas/Silverpeas-Core/blob/246dfaf16870c9378ceccddc4402712598652e62/core-war/src/main/java/org/silverpeas/web/jobmanager/servlets/JobManagerPeasRequestRouter.java#L76): The Cyclomatic Complexity of this method "getDestination" is 11 which is greater than 10 authorized. [rule](http://integration.silverpeas.org/sonar/coding_rules#rule_key=squid%3AMethodCyclomaticComplexity)

[MAJOR]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR'
[MINOR]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR'

**Comment by silveruser** [comment]
SonarQube analysis reported 1 issue

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [JobManagerPeasRequestRouter.java#L76](https://github.com/Silverpeas/Silverpeas-Core/blob/34714d10b7b52f6e925e22d0a0a81f0b76d40559/core-war/src/main/java/org/silverpeas/web/jobmanager/servlets/JobManagerPeasRequestRouter.java#L76): The Cyclomatic Complexity of this method "getDestination" is 11 which is greater than 10 authorized. [rule](http://integration.silverpeas.org/sonar/coding_rules#rule_key=squid%3AMethodCyclomaticComplexity)

[MAJOR]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR'

**Comment by SilverYoCha** [comment]
Merged.
Closed manually because of manual git operations performed during integration.

**Pull request closed without merging**